src/middleware/APIAuth.ts

```diff
@@ -0,0 +1,81 @@
+import axios from 'axios';
+
+/*
+ *   This class contains express middleware for endpoints with the following format:
+ *       /log?project=[projectID]
+ *
+ *   The request must have the header 'log_key' with a 10-character
+ *   string value. This middleware will be to validate the key based on
+ *   the user specified in the endpoint's stored key in the DB.
+ *
+ *   This method will only be secure once the endpoints established in the
+ *   web app backend are protected.
+ *
+ *   Constructor args must include the URI to the webapp,
+ *   during development it is localhost:3000
+ */
+export default class AuthVerification {
+    private gateURI: string;
+
+    // passed in through req.query
+    private projectID: string;
+
+    // passed in through req.headers
+    private logKey: string;
+
+    constructor(gateURI: string, projectID: string, logKey: string) {
+        this.gateURI = gateURI;
```

Comment:
I don't know why `gateURI` has to be passed in through constructor when it is the same for every implementation of logger. It could just be hard coded and eliminate the chance of developer error.